Connectivity guide update

Gauravu2 opened the pull request:
Code for Granger Causality (Pairwise), Power and cross power spectral density, and welch coherence are added

skim0119 reviewed (changes requested):
1. I updated how to setup the workspace [here](https://github.com/skim0119/MiV-OS/blob/patch_contribution/CONTRIBUTING.md#before-i-get-start-contributing). Please have it setup. You'll also need extra installation for documentation.
2. Take a look at [this one](https://github.com/skim0119/MiV-OS/blob/patch_contribution/docs/README.md) as well, especially second point in `key things to remember`.
3. Add some descriptions and references inside the documentation as well. ([example](https://miv-os.readthedocs.io/en/doc_patch/guide/signal_processing.html))
4. Rename the file: `granger_causality_psd_cpsd_coherence`. (no capital)
5. Plotting function can be moved to `miv/visualization/connectivity.py` (or you name it). Take a look at other files in the same folder for examples.

Gauravu2 commented:
Added functions for causality, causality plots and spectral plots. Updated the init files in statistics and visualization. The documentation is added to docs/guide

skim0119 reviewed (approved):
LGTM. @Gauravu2 I made some changes in the documentation. Please take a look.
